docs/experiments-fever.md

```diff
@@ -0,0 +1,196 @@
+```
+
+## Comparing with FEVER Baseline
+
+We can also evaluate our retrieval compared to the TF-IDF baseline described in [the FEVER paper](https://www.aclweb.org/anthology/N18-1074.pdf). Specifically, we want to compare the metrics described in Table 2 of the paper.
+
+We evaluate the run file produced earlier:
+
+```bash
+python src/main/python/fever/evaluate_doc_retrieval.py \
+ --truth_file collections/fever/paper_dev.jsonl \
+ --run_file runs/run.fever-paragraph.dev.tsv
+```
+
+This run produces the following results:
+
+| k   | Fully Supported | Oracle Accuracy |
+|:----|----------------:|----------------:|
+| 1	  | 0.3271827183    | 0.5514551455    |
+| 5	  | 0.5655565557    | 0.7103710371    |
+| 10  |	0.6542154215    | 0.7694769477    |
+| 25  |	0.7458745875    | 0.8305830583    |
+| 50  |	0.8097809781    | 0.8731873187    |
+| 100 |	0.8561356136    | 0.9040904090    |
+
+### BM25 Tuning
+
+The above retrieval uses the MS MARCO default BM25 parameters of `k1=0.82`, `b=0.68`. We can tune these parameters to outperform the results of the TF-IDF baseline in the paper.
+
+We tune on a subset of the training split of the dataset. We generate that subset:
+
+```bash
+python src/main/python/fever/generate_subset.py \
+ --dataset_file collections/fever/train.jsonl \
+ --subset_file collections/fever/train-subset.jsonl
+```
+
+We then generate the queries and qrels files for this subset.
+
+```bash
+python src/main/python/fever/generate_queries_and_qrels.py \
+ --dataset_file collections/fever/train-subset.jsonl \
+ --output_queries_file collections/fever/queries.paragraph.train-subset.tsv \
+ --output_qrels_file collections/fever/qrels.paragraph.train-subset.tsv \
+ --granularity paragraph
+```
+
+We tune the BM25 parameters with a grid search of parameter values in 0.1 increments. We save the run files generated by this process to a new folder `runs/fever-bm25` (do not use `runs` here).
+
+```bash
+python src/main/python/fever/tune_bm25.py \
+ --runs_folder runs/fever-bm25 \
+ --index_folder indexes/fever/lucene-index-fever-paragraph \
+ --queries_file collections/fever/queries.paragraph.train-subset.tsv \
+ --qrels_file collections/fever/qrels.paragraph.train-subset.tsv
+```
+
+From the grid search, we observe that the parameters `k1=0.6`, `b=0.5` perform fairly well. If we retrieve on the dev set with these parameters:
+
+```bash
+python tools/scripts/msmarco/retrieve.py \
+ --hits 1000 --threads 1 \
+ --index indexes/fever/lucene-index-fever-paragraph \
+ --queries collections/fever/queries.paragraph.dev.tsv \
+ --output runs/run.fever-paragraph-0.6-0.5.dev.tsv \
+ --k1 0.6 --b 0.5
+```
+
+and we evaluate this run file:
+
+```bash
+python src/main/python/fever/evaluate_doc_retrieval.py \
+ --truth_file collections/fever/paper_dev.jsonl \
+ --run_file runs/run.fever-paragraph-0.6-0.5.dev.tsv
+```
+
+then we can achieve the following results:
+
+| k   | Fully Supported | Oracle Accuracy |
+|:----|----------------:|----------------:|
```

Comment:
Can we fix this at the same sig figs as the table in the paper (modify the code to do that too?). Table will look a little neater.

Reply:
My concern for modifying the code to do that is that the tabbed printing will get messed up, but I guess that's not too important lol